### PR TITLE
Directly convert tweet id to timestamp

### PR DESCRIPTION
**Pull request by perryizgr8**
This will work even if twitter element is blocked by tracking
protection.

### CODE REVIEW

**Comment by perryizgr8** [comment]
Just realized that this may need a bit of an explanation. Twitter uses something called snowflake to generate the tweet IDs. It encodes the timestamp in it. Full details at https://github.com/client9/snowflake2time. I looked at the function snowflake2utc() from the same repo: https://github.com/client9/snowflake2time/blob/master/python/snowflake.py and implemented it in js. Apparently js does not support 64 bit integers fully so right shift does not work properly. But division works. Right shift by 22 bits translates to division by 4194304 (2^22).

So basically we can directly extract the timestamp of a tweet from the id. So no need to wait for the tweets to actually load. This should even work if the embedded elements are blocked by tracking protection add-ons.

File: index.html [hunk 1]
```diff
@@ -51,29 +51,22 @@ <h2>Archive:</h2>
         twttr.widgets.createTweet(archiveTweetId, tweetDiv, {});
       });
 
+      // Use latest tweet id to get the timestamp
+      var date = new Date(Math.floor(String(Math.floor(latestTweetId / 4194304))) + 1288834974657);
```
Comment (stryju) [inline index.html]: why the double `Math.floor` + casting to string?

wouldn't this suffice?
```
new Date(Math.floor(latestTweetId / 4194304) + 1288834974657);
```

File: index.html [hunk 2]
```diff
@@ -51,29 +51,22 @@ <h2>Archive:</h2>
...
+
+      // Subtract from now and convert to days (from milliseconds)
+      var daysAgo = (new Date - date) / 1000 / 60 / 60 / 24;
```
Comment (stryju) [inline index.html]: since you're working with ints now, why use `Date` at all? :)

File: index.html (hunk 1 above)
Comment (perryizgr8) [reply index.html]: Yes you're right, this should be enough. I am doing github pull requests for the first time, how do I change the line here? Should I close this PR and submit a new one? Thanks for the review.

File: index.html (hunk 2 above)
Comment (perryizgr8) [reply index.html]: How else can I get the current timestamp in milliseconds? Even if I use something else like getTime() isn't that the same thing as this?

File: index.html (hunk 1 above)
Comment (perryizgr8) [reply index.html]: Figured it out :)

**Comment by harryh** [comment]
This is great. Thanks!

File: index.html (hunk 2 above)
Comment (stryju) [reply index.html]: `Date.now()` seems OK
also, subtracting a number from a `Date` would cast the `Date` to a number (so it would return `ms`) :)

Comment (stryju) [reply index.html]: so, this should do just fine:

```js
// Use latest tweet id to get the timestamp
var date = Math.floor(latestTweetId / 4194304) + 1288834974657;

// Subtract from now and convert to days (from milliseconds)
var daysAgo = (Date.now() - date) / 1000 / 60 / 60 / 24;
```

Comment (harryh) [reply index.html]: @stryju: if you look at master you'll see that after merging in @perryizgr8's change I essentially made your suggestions in a subsequent commit
